merkledag/cidversion.go

```diff
@@ -0,0 +1,26 @@
+package merkledag
+
+import (
+	"fmt"
+)
+
+// CidVersion is a a cid version number that is guaranteed to be valid
+type CidVersion struct {
```

Comment:
The idea was to provide some additional type safety so once you have a CidVersion you don't need to return an error if the version is invalid as it has already been checked.